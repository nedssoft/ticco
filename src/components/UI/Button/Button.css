.Button {
  background-color: transparent;
  border: none;
  color: white;
  outline: none;
  cursor: pointer;
  font: inherit;
  padding: 10px;
  margin: 10px;
  font-weight: bold;
}

.Button:first-of-type {
  margin-left: 0;
  padding-left: 0;
}

.Success {
  color: #5C9210;
}

.Danger {
  color: #944317;
}
.Button:disabled {
  color: #ccc;
  cursor: not-allowed;
}
.btnDefault{
  padding: 10px 0 10px 0;
  text-align: center;
  line-height: 20px;
  background-color: #EE6860;
  letter-spacing: 0.3px;
  color: aliceblue;
  border: none;
  outline: none;
  font-size: 1em;
  font-weight: 600;
  text-decoration: none;
  cursor: pointer; 
}