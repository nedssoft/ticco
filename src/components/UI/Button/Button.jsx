import React from 'react'
import classes from './Button.css'

const Button = ({ clicked, children, btnType, disabled}) => {
  return (
    <button 
      onClick={clicked}
      disabled={disabled}
      className={[classes.Button, classes[btnType]].join(' ')}
    >
      { children }
    </button>
      
  )
}

export default Button
