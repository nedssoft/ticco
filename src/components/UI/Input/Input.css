.Input {
  width: 100%;
}
.Label {
  font-weight: bold;
  display: block;
  margin-bottom: 5px;
  box-sizing: border-box;
  text-align: left;
  width: 100%;
  margin-left: 10%;
}
.InputElement{
  outline: none;
  width: 80%;
  border: 1px solid #ccc;
  background: #fff;
  padding: 6px 10px;
  font: inherit;
  display: block;
  margin: 10px auto;
  font-size: 18px;
  border-radius: 3px;
}

.InputElement:focus {
  outline: none;
  background: #ccc;
}

.Invalid {
  border: 1px solid #bb4430;
  background-color: #bb4430;
}

.Valid {
  border: 1px solid rgb(106, 161, 106)
}
