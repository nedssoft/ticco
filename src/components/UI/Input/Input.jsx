import React from 'react'

import styles from './Input.css'


const Input = (props) => {
  const { 
    elementType, 
    elementConfig, 
    value, 
    label, 
    changed,
    valid,
    shouldValidate,
    touched
  } = props;
  const { id } = elementConfig
  let inputElement = null
  const inputClasses = [styles.InputElement]

  if (!valid && shouldValidate && touched) {
    inputClasses.push(styles.Invalid)
  }
  if (valid) {
    inputClasses.push(styles.Valid)
  }
  switch (elementType) {
    case 'input':
      inputElement = (
        <input 
          {...elementConfig}
          className={inputClasses.join(' ')}
          onChange={changed}
          value={value}
          id={id}
        />
      );
      break;
    case 'textarea':
      inputElement = (
        <textarea 
          {...elementConfig} 
          className={inputClasses.join(' ')} 
          onChange={changed}
          value={value}
          id={id}
        />
      )
      break;
      case 'select':
      inputElement = (
        <select 
          className={inputClasses.join(' ')} 
          onChange={changed}
          value={value}
          id={elementConfig.id}
          label={elementConfig.label}
        >
          {elementConfig.options.map(option => (
            <option value={option.value} key={option.value}>
              {option.displayValue}
            </option>
          ))} 
        </select>
      )
      break;
    default:
      inputElement = (
        <input 
          {...elementConfig} 
          className={inputClasses.join(' ')} 
          onChange={changed}
          value={value}
        />
      )
}
  return (
    <div className={styles.Input}>
      <label htmlFor={id} className={styles.Label}> {label}</label>
      { inputElement }
    </div>
  )
}

export default Input
