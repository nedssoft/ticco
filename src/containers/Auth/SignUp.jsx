import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'
import styles from './Auth.css'
import Spinner from '../../components/UI/Spinner/Spinner'
import Input from '../../components/UI/Input/Input'
import Button from '../../components/UI/Button/Button'
import { checkValidity } from '../../helpers/helper'
import { successFeedback, errorFeedback } from '../../helpers/FeedbackMessage'
import { registerUser } from '../../store/actions'

class SignUp extends Component {
  state = {
    controls: {
      firstName: {
        elementType: 'input',
        elementConfig: {
          placeholder: 'Your First Name',
          id: 'firstName',
          type:'text'
        },
        value: '',
        label: 'First Name',
        validation: {
          required: true
        },
        valid: false,
        shouldValidate: true
      },
      lastName: {
        elementType: 'input',
        elementConfig: {
          placeholder: 'Your Last Name',
          id: 'lastName',
          type:'text'
        },
        value: '',
        label: 'Last Name',
        validation: {
          required: true
        },
        valid: false,
        shouldValidate: true
      },
      email:{
        elementType: 'input',
        elementConfig: {
          placeholder: 'Your E-mail',
          id: 'email',
          type:'email'
        },
        value: '',
        label: 'Email',
        validation: {
          required: true,
          isEmail: true
        },
        valid: false,
        shouldValidate: true
      },
      phone:{
        elementType: 'input',
        elementConfig: {
          placeholder: 'Your Phone Number',
          id: 'phone',
          type:'tel'
        },
        value: '',
        label: 'Phone',
        validation: {
          required: true,
          minLength: 11
        },
        valid: false,
        shouldValidate: true
      },
      password:{
        elementType: 'input',
        elementConfig: {
          placeholder: 'Password',
          id: 'password',
          type:'password'
        },
        value: '',
        label: 'Password',
        validation: {
          required: true,
          minLength: 6
        },
        valid: false,
        shouldValidate: true
      },
    },
    formIsValid: false
  }
  inputChangeHandler = (event, inputId) => {
    const { controls } = this.state
    const updatedControls = { ...controls }
    const updatedElement = { ...updatedControls[inputId]}
    updatedElement.value = event.target.value;
    updatedElement.touched = true;
    updatedElement.valid = checkValidity(updatedElement.value , updatedElement.validation);
    updatedControls[inputId] = updatedElement;

    let formIsValid = true;
    for (let key in updatedControls) {
      formIsValid = updatedControls[key].valid && formIsValid
    }
    this.setState({controls: updatedControls, formIsValid});
  }
  submitHandler = (event) => {
    event.preventDefault()
    const { controls, formIsValid } = this.state;
    if(!formIsValid) {
      errorFeedback('Fix the errors indicated')
      return false
    }
    const userData = this.extractFormData(controls)
    this.props.registerUser(userData)
  }
  extractFormData = (data) => {
    let formData = {}
    for (let key in data) {
      formData[key] = data[key].value;
    }
    return formData;
  }
  render() {
    const { controls, formIsValid } = this.state
    let formElementArr = [];
    for (let key in controls) {
      formElementArr.push({
        id: key,
        config: controls[key]
      });
    }
    return  (
      <div className={styles.Container}>
        <div className={styles.Auth}>
          { this.props.isLoading ? <Spinner /> : ''}
          <form onSubmit={this.submitHandler}>
            { formElementArr.map(element => {
          return (
            <Input 
              elementType={element.config.elementType}
              elementConfig={element.config.elementConfig}
              value={element.config.value}
              label={element.config.label}
              key={element.id}
              valid={element.config.valid}
              shouldValidate={element.config.shouldValidate}
              touched={element.config.touched}
              changed={(event) => this.inputChangeHandler(event, element.id)}
            />
          )
        })}
            <Button disabled={!formIsValid} btnType="Submit">Sign Up</Button>
          </form>
          <Link to="/signin"> Already Registered? Sign in</Link>
        </div>
      </div>
)
  }
}
const mapStateToProps = state => {
  return {
    isLoading: state.auth.isLoading,
    isLoggedIn: state.auth.token !== null || localStorage.getItem('token')
  }
}
export default connect(mapStateToProps, { registerUser})(SignUp)
