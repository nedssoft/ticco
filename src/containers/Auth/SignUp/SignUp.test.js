import React from 'react';
import { shallow } from 'enzyme'
import { SignUp } from './SignUp'
import Button from '../../../components/UI/Button/Button'
import Input from '../../../components/UI/Input/Input'

describe('Auth Components', () => {
 it('should render sign up page', () => {
  const wrapper = shallow(<SignUp />)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('form')).toHaveLength(1)
    expect(wrapper.find(Button)).toHaveLength(1)
    expect(wrapper.find(Input)).toHaveLength(5)
 })
})