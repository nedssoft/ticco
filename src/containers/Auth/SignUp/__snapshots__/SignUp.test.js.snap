// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth Components should render sign up page 1`] = `
<div
  className="Container"
>
  <div
    className="Auth"
  >
    <form
      onSubmit={[Function]}
    >
      <Input
        changed={[Function]}
        elementConfig={
          Object {
            "id": "firstName",
            "placeholder": "Your First Name",
            "type": "text",
          }
        }
        elementType="input"
        key="firstName"
        label="First Name"
        shouldValidate={true}
        valid={false}
        value=""
      />
      <Input
        changed={[Function]}
        elementConfig={
          Object {
            "id": "lastName",
            "placeholder": "Your Last Name",
            "type": "text",
          }
        }
        elementType="input"
        key="lastName"
        label="Last Name"
        shouldValidate={true}
        valid={false}
        value=""
      />
      <Input
        changed={[Function]}
        elementConfig={
          Object {
            "id": "email",
            "placeholder": "Your E-mail",
            "type": "email",
          }
        }
        elementType="input"
        key="email"
        label="Email"
        shouldValidate={true}
        valid={false}
        value=""
      />
      <Input
        changed={[Function]}
        elementConfig={
          Object {
            "id": "phone",
            "placeholder": "Your Phone Number",
            "type": "tel",
          }
        }
        elementType="input"
        key="phone"
        label="Phone"
        shouldValidate={true}
        valid={false}
        value=""
      />
      <Input
        changed={[Function]}
        elementConfig={
          Object {
            "id": "password",
            "placeholder": "Password",
            "type": "password",
          }
        }
        elementType="input"
        key="password"
        label="Password"
        shouldValidate={true}
        valid={false}
        value=""
      />
      <Button
        btnType="Submit"
        disabled={true}
      >
        Sign Up
      </Button>
    </form>
    <Link
      to="/signin"
    >
       Already Registered? Sign in
    </Link>
  </div>
</div>
`;
